In [ ]:
#KATHERINE FLORES - HOMEWORK 4

In [37]:
# 13.4
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols

df = pd.read_csv('/content/SQC_HW4_data.csv')

rep1 = df.copy()
rep1['Y'] = df['Response']
rep2 = df.copy()
rep2['Y'] = df['Response2']
df_long = pd.concat([rep1, rep2], ignore_index=True)

model = ols('Y ~ A + B + C + A:B + A:C + B:C + A:B:C', data=df_long).fit()
aov_table = sm.stats.anova_lm(model, typ=2)
print("\n ANOVA Table ")
print(aov_table)

print(model.summary())



 ANOVA Table 
            sum_sq   df          F    PR(>F)
A          1332.25  1.0   0.541015  0.483017
B         28392.25  1.0  11.529848  0.009422
C         20592.25  1.0   8.362335  0.020144
A:B         506.25  1.0   0.205584  0.662297
A:C       56882.25  1.0  23.099391  0.001345
B:C        2352.25  1.0   0.955228  0.357017
A:B:C      4830.25  1.0   1.961523  0.198923
Residual  19700.00  8.0        NaN       NaN
                            OLS Regression Results                            
Dep. Variable:                      Y   R-squared:                       0.854
Model:                            OLS   Adj. R-squared:                  0.726
Method:                 Least Squares   F-statistic:                     6.665
Date:                Wed, 15 Oct 2025   Prob (F-statistic):            0.00790
Time:                        06:09:18   Log-Likelihood:                -79.629
No. Observations:                  16   AIC:                             175.3
Df Residuals:              

13.4
The R^2 is 0.854 so that means that the model explains 85.4% of the total variation in the response of tool life. The adjusted R^2 lowers down to 0.726 so that means after penalizing for extra predictors, our starts fitting a little less correctly so there may be factors that aren't really significant. Our F-Statistic is 6.665 which is not too high but because our p value is 0.007 our model is statistically significant. Now to make sure our model assumptions are met we look at the diagnostics. First is omnibus and beacsue our p value is 0.570 we are in a normal range. Our durbin watson is 1.54 which is acceptable. Our Con No. is 1.00 so this is excellent.
a)My final conclusion is that factor b and c significantly affect tool life because of their low p value. Factor b has p value of 0.009 and a coefficient of 42 which is very high. Even at a 99% confidence level we can say that this factor is statistically significant. Factor c is also very significant with a coefficient of 35 and a p value of 0.02, at a confidence level of .98 we can conclude that this factor is statistically significant. We cannot say the saem baout factor a, because it has little to no correlation with tool life.
b) The pattern I am observing is that once factors start being intercepted or both being incorporated then it actually decreases the coefficient and it ends up being negative. This means the intersecitons decrease tool life. Factors a an c combined most strongly decrease tool life. I cannot conclude any of the combination of factors produces a longer tool life than individual factors themselves.
c)I would say the best combination of cutting speed and cutting angle that always gives good results regardless of metal hardness is when cutting speed is at a high and cutting angle is at a high, it doesn't really matter the hardness of the metal because ac is our highest and most positive correlation.

In [42]:
#13.6

import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols


df = pd.read_csv('/content/SQC_HW_data2.csv')

rep1 = df.copy()
rep1['Y'] = df['Response1']

rep2 = df.copy()
rep2['Y'] = df['Response2']

df_long = pd.concat([rep1, rep2], ignore_index=True)


model = ols('Y ~ A + B + C + D + A:B + A:C + A:D + B:C + B:D + C:D + \
            A:B:C + A:B:D + A:C:D + B:C:D + A:B:C:D', data=df_long).fit()


aov_table = sm.stats.anova_lm(model, typ=2)
print("\n===== ANOVA Table =====")
print(aov_table)


print("\n===== Model Summary =====")
print(model.summary())



===== ANOVA Table =====
             sum_sq    df          F    PR(>F)
A         657.03125   1.0  22.729730  0.000210
B          13.78125   1.0   0.476757  0.499790
C          57.78125   1.0   1.998919  0.176574
D         124.03125   1.0   4.290811  0.054847
A:B       132.03125   1.0   4.567568  0.048369
A:C         3.78125   1.0   0.130811  0.722325
A:D        38.28125   1.0   1.324324  0.266726
B:C         2.53125   1.0   0.087568  0.771099
B:D         0.28125   1.0   0.009730  0.922650
C:D        22.78125   1.0   0.788108  0.387824
A:B:C     215.28125   1.0   7.447568  0.014861
A:B:D     175.78125   1.0   6.081081  0.025348
A:C:D       7.03125   1.0   0.243243  0.628574
B:C:D       7.03125   1.0   0.243243  0.628574
A:B:C:D    47.53125   1.0   1.644324  0.218001
Residual  462.50000  16.0        NaN       NaN

===== Model Summary =====
                            OLS Regression Results                            
Dep. Variable:                      Y   R-squared:                    

13.6
For this analysis I checked the R^2 which is 0.765 but our adjusted R^2 is 0.545 which means after penalizing for extra fluff in our model, our model is less statistically significant. Overall our model explains 76.5% of the total variation in taste of soft-drink beverage. Our F-statistic is 3.472 and our p value is 0.009 so even at a 99% confidence interval we are sure our model is statistically significant. To make sure our model assumptions are met we checked the diagnostics. Omnibus is 0.0000 so thats really good and Durbin Watson is 1.8 which is high but acceptable. and our Cond No is 1.00 so that is excellent. After looking at the p values for all the factors, I am concluding that factor a is statistically significant with a p value of 0.000 so we can definately conclude it explains a lot in our model. Also our coefficient for that factor is -4 which is very high. This factor is type of sweetener. The pattern I am seeing is that AB, ABC, and ABD have the lowest p values that would be statistically significant at a 95% confidence level. Overall type of sweetener is the most statistically significant factor in our model explaining most of our variation in the model.